packages/web/src/app/api/(server)/commits/route.ts

```diff
@@ -0,0 +1,27 @@
+import { searchCommits, SearchCommitsRequest } from "@/features/search/gitApi";
+import { serviceErrorResponse, schemaValidationError } from "@/lib/serviceError";
+import { isServiceError } from "@/lib/utils";
+import { NextRequest } from "next/server";
+import { searchCommitsRequestSchema } from "@/features/search/schemas";
+import { withOptionalAuthV2 } from "@/withAuthV2";
+
+export async function POST(request: NextRequest): Promise<Response> {
+    const body = await request.json();
+    const parsed = await searchCommitsRequestSchema.safeParseAsync(body);
+
+    if (!parsed.success) {
+        return serviceErrorResponse(
+            schemaValidationError(parsed.error)
+        );
+    }
+
+    const result = await withOptionalAuthV2(async () => {
+        return await searchCommits(parsed.data as SearchCommitsRequest);
+    });
```

Comment:
Nit:
1. The `as` assertion is unnecessary because the `safeParseAsync` guards the type.
2. The `async` and `await` in the callback functions are also unnecessary because the `searchCommits` is the only function you are calling here, so you are not doing any synchronous calls.
3. The `withOptionalAuthV2` is redundant here because you already have it in the `searchCommits` method. 

```suggestion
    const result = await searchCommits(parsed.data);
```

Reply:
Done, thanks